wandb/sdk/launch/runner/kubernetes_runner.py

```diff
@@ -99,6 +108,110 @@ def __init__(
         self._fail_count = 0
         self.pod_names = pod_names
         self.secret = secret
+        self._status_lock = Lock()
```

Comment:
we should store the latest `resourceVersion` we've seen so our watches can be efficient: https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes

Reply:
sdk is also doing this behind the scenes :)

Reply:
I think what Jason is saying is if we maintain the state of the `resourceVersion` we can request the stream of events for that resource starting from the resourceVersion. This makes it so we never miss anything

Reply:
I think the k8s sdk is doing this for us already: https://github.com/kubernetes-client/python/blob/76ec6f5ce5572f66d82717e296480dc42584f0b7/kubernetes/base/watch/watch.py#L194C36-L194C36